Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -194,12 +196,86 @@ void AODProducerWorkflowDPL::addToMFTTracksTable(mftTracksCursorType& mftTracksC
 }
 
+template <typename TracksCursorType, typename TracksCovCursorType, typename TracksExtraCursorType, typename mftTracksCursorType>
+void AODProducerWorkflowDPL::fillTrackTablesPerCollision(int collisionID,
+                                                         const o2::dataformats::VtxTrackRef& trackRef,
+                                                         gsl::span<const GIndex>& GIndices,
+                                                         o2::globaltracking::RecoContainer& data,
+                                                         TracksCursorType& tracksCursor,
+                                                         TracksCovCursorType& tracksCovCursor,
+                                                         TracksExtraCursorType& tracksExtraCursor,
+                                                         mftTracksCursorType& mftTracksCursor)
+{
+  const auto& tpcClusRefs = data.getTPCTracksClusterRefs();
+  const auto& tpcClusShMap = data.clusterShMapTPC;
+  const auto& tpcClusAcc = data.getTPCClusters();
+
+  for (int src = GIndex::NSources; src--;) {
+    int start = trackRef.getFirstEntryOfSource(src);
+    int end = start + trackRef.getEntriesOfSource(src);
+    LOG(DEBUG) << "Unassigned tracks: src = " << src << ", start = " << start << ", end = " << end;
+    for (int ti = start; ti < end; ti++) {
+      TrackExtraInfo extraInfoHolder;
+      auto& trackIndex = GIndices[ti];
+      if (GIndex::includesSource(src, mInputSources)) {
+        if (src == GIndex::Source::MFT) { // MFT tracks are treated separately since they are stored in a different table
+          const auto& track = data.getMFTTrack(trackIndex.getIndex());
+          addToMFTTracksTable(mftTracksCursor, track, collisionID);
+        } else {
+          auto contributorsGID = data.getSingleDetectorRefs(trackIndex);
+          const auto& trackPar = data.getTrackParam(trackIndex);
+          if (contributorsGID[GIndex::Source::ITS].isIndexSet()) {
+            const auto& itsOrig = data.getITSTrack(contributorsGID[GIndex::ITS]);
+            extraInfoHolder.itsClusterMap = itsOrig.getPattern();
+          }
+          if (contributorsGID[GIndex::Source::TPC].isIndexSet()) {
+            const auto& tpcOrig = data.getTPCTrack(contributorsGID[GIndex::TPC]);
+            extraInfoHolder.tpcInnerParam = tpcOrig.getP();
+            extraInfoHolder.tpcChi2NCl = tpcOrig.getNClusters() ? tpcOrig.getChi2() / tpcOrig.getNClusters() : 0;
+            extraInfoHolder.tpcSignal = tpcOrig.getdEdx().dEdxTotTPC;
+            uint8_t shared, found, crossed; // fixme: need to switch from these placeholders to something more reasonable
+            countTPCClusters(tpcOrig, tpcClusRefs, tpcClusShMap, tpcClusAcc, shared, found, crossed);
+            extraInfoHolder.tpcNClsFindable = tpcOrig.getNClusters();
+            extraInfoHolder.tpcNClsFindableMinusCrossedRows = tpcOrig.getNClusters() - crossed;
+            extraInfoHolder.tpcNClsShared = shared;
```

Comment:
There are 4 variables to fill, I would use:
````
extraInfoHolder.tpcNClsFindable = tpcOrig.getNClusters();
extraInfoHolder.tpcNClsFindableMinusFound = tpcOrig.getNClusters() - found;
extraInfoHolder.tpcNClsFindableMinusCrossedRows = tpcOrig.getNClusters() - crossed;
extraInfoHolder.tpcNClsShared = shared;
````
bearing in mind that such a definition of `tpcNClsFindable` is wrong and in some cases the `tpcOrig.getNClusters() - crossed` can be <0. Later we should find a proper substitute for the tpcNClsFindable.
As I wrote before, the `tpcOrig.getNClusters()` itself is not a good variable, since in the new reconstruction if the fitter considers that the 2 clusters on the same padrow may come in fact from the same split cluster, it will count (and fit) both of them, thus the `getNClusters()` may actually exceed the number of crossed rows.

Reply:
Done, thank you